### ISSUE TITLE
🐛 Bug Report: Notification fails due to subscriber not having channel configured for push channel

### ISSUE THREAD

**Issue by jarredwitt** [novuhq/novu#4960]
### 📜 Description

We have a workflow that sends out an email and a push notification to all of our subscribers. Push notifications are opt-in on our platform so while the subscriber will have an email channel they may not have the push channel configured if they didn't opt-in. This leads to a lot of our notifications displaying like this in the activity feed:

![CleanShot 2023-12-08 at 10 18 52](https://github.com/novuhq/novu/assets/5090577/7c310cbb-653d-4599-9800-f5adcb2a4243)

IMO the notification didn't fail to send, the subscriber just didn't have the channel configured. Is it possible to skip a workflow step if a channel isn't configured for a subscriber? This is really bloating our activity feed with lots of failures when in fact they aren't actually failing. Our worker logs are also displaying a lot of the same errors and it's causing a lot of noise in our logging for no reason.

### 👟 Reproduction steps

Create workflow that has an email step and a push notification step. Send a notification to a subscriber that has the email channel configured, but not the push notification channel.

### 👍 Expected behavior

It should skip the push notification step if the subscriber doesn't have the channel configured. 

### 👎 Actual Behavior with Screenshots

The notification fails due to the channel not being configured. 

![CleanShot 2023-12-08 at 10 18 52](https://github.com/novuhq/novu/assets/5090577/02fbd9df-a778-45a6-baa4-1a1af3829edd)

### Novu version

0.21.0

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find a similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

**Comment by uncle-tee** [comment]
@jainpawan21 can I pick this for grabs, it's my first time here and I want to see if this is possible to achieve. Let me know if you want me to pick this.

**Comment by jainpawan21** [comment]
Hi @uncle-tee 

Thanks for your interest in contribution. As this github issue is suggesting a change in how execution works, I will need to discuss this with team if we should go ahead as suggested in this issue.

**Comment by Pornpunnarai** [comment]
How to fixed this issue?

**Comment by paulwer** [comment]
Could this at least be treated different in the ui?

Some note: This also results in an issue where the frontend stops refreshing the status of the workflow in the local studio while executing a test run. Steps afterwards get executed, but this is not visible in the frontend.

**Comment by jarredwitt** [comment]
Is there a way this can be elevated? Our admin activity feed shows 400+ notification errors for this month because of this bug. It's creating a ton of false positives for us and additional work for our admins because we have to look at each failure to see if there is actually a problem.

**Comment by Reizar** [comment]
+1 this is also causing big issues for us. 

It also means we can't use the retry logic. We have the same setup where users have emails on by default and push notifications are opt in. We can't use the retry logic because even though the email step in the workflow succeeds, the push notification step fails and then it retries the entire workflow. Which causes emails to be sent over and over again.

**Comment by bibhuti9** [comment]
Hey any solution?

**Comment by scopsy** [comment]
This was resolved in the latest api versions, and now those steps will be marked as skipped

**Comment by paulwer** [comment]
@scopsy i dont think this is fixed, can you please take a look

bug present at 3.11.0 and 3.12.0

<img width="1557" height="805" alt="Image" src="https://github.com/user-attachments/assets/aa3be82b-4753-4c74-b756-aa40dc6caeaa" />

**Comment by paulwer** [comment]
@scopsy was this fixed?

**Comment by scopsy** [comment]
Yes, just merged, pending deployment and should be available in the upcoming minor release for OSS 🙏

**Comment by paulwer** [comment]
@scopsy thank yall very much❤️✌️
